include/swift/Driver/DependencyGraph.h

```diff
@@ -263,6 +263,31 @@ class DependencyGraph : public DependencyGraphImpl {
                                                data);
   }
 
+  LoadResult loadFromString(T node, StringRef key, StringRef data) {
+    return DependencyGraphImpl::loadFromString(Traits::getAsVoidPointer(node),
+                                               key.str() + data.str());
+  }
+
+  LoadResult loadFromString(T node,
+                            StringRef key1, StringRef data1,
+                            StringRef key2, StringRef data2) {
+    return DependencyGraphImpl::loadFromString(Traits::getAsVoidPointer(node),
+                                               key1.str() + data1.str() +
+                                               key2.str() + data2.str());
+  }
+
+  LoadResult loadFromString(T node,
+                            StringRef key1, StringRef data1,
+                            StringRef key2, StringRef data2,
+                            StringRef key3, StringRef data3,
+                            StringRef key4, StringRef data4) {
+    return DependencyGraphImpl::loadFromString(Traits::getAsVoidPointer(node),
+                                               key1.str() + data1.str() +
+                                               key2.str() + data2.str() +
+                                               key3.str() + data3.str() +
+                                               key4.str() + data4.str());
+  }
```

Comment:
I'm not sure this is an improvement over just appending strings, but if you're set on it please make them helper functions in the test file rather than changing the actual type. No one will ever need this functionality for real code.

Reply:
Sure, done!